adagrams/game.py

```diff
@@ -1,11 +1,155 @@
+        letter_index = random.randint(0, len(LETTER_POOL) -1)
+
+        if letters_frequency[letter_index] > 0:
+            letters_frequency[letter_index] -= 1
+            final_string.append(letters_list[letter_index])
+        
+    return final_string
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    available_letters = []
+
+
+    for letter_item in letter_bank:
+        available_letters.append(letter_item)
+
+    for letter in word:
+        uppercase_letter = letter.upper()
+        if uppercase_letter in available_letters:
+            
+            available_letters.remove(uppercase_letter)
+        else: 
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    
+    score_values = {
+        1: ["A", "E", "I", "O", "U", "L", "N", "R", "S", "T"],
+        2: ["D", "G"],
+        3: ["B", "C", "M", "P"],
+        4: ["F", "H", "V", "W", "Y"],
+        5: ["K"],
+        8: ["J", "X"],
+        10: ["Q", "Z"]
+    }
+    score = 0
+    index = 0
+    word_uppercase = word.upper()
+    for letter in word_uppercase:
+        for num_key, letter_value in score_values.items():
+            if letter in letter_value:
+                score += num_key
+    if  6 < len(word_uppercase) < 11:
+        score += 8
+    print(score)
+    return score
+
 
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    words_entered = []
+    score_data = []
+    index = 0
+    
+    for word in word_list:
+        score = score_word(word)
+        score_data.append(score)
+        words_entered.append(word)
+
+    winning_score =  score_data[0]
+    winning_word = words_entered[0]
+
+    for score in score_data:
+        
+        if score > winning_score:
+            winning_score = score_data[index]
+            winning_word = words_entered[index]
+        elif score == winning_score:
+            if len(winning_word) == 10:
+                continue
+            elif len(words_entered[index]) == 10:
+                winning_score = score_data[index]
+                winning_word = words_entered[index]
+            elif len(words_entered[index]) < len(winning_word) :
+                winning_score = score_data[index]
+                winning_word = words_entered[index]
+            elif len(winning_word) > len(words_entered[index]):
+                continue
+            else:
+                
+                loop_count = 0
+                for index_word in words_entered:
+                    if index_word == winning_word:
+                        winning_score_index = loop_count
+                    else:
+                        loop_count += 1
+
+                if index < winning_score_index:
+                    winning_score = score_data[index]
+                    winning_word = words_entered[index]
+                
+        if index == len(score_data):
+            return (winning_word, winning_score)
+        else:
+            
+            index += 1
```

Comment:
So one thing here complicating your logic is that it needs to keep track of the indexes in order to return the correct score and word. We could simplify this by writing code that is index agnostic. Consider if we calculated a score of a word and in that same iteration we decided to do the tie breaking logic as well. It could look a little something like this: 

```python
def get_highest_word_score(word_list):
    highest_score = 0
    winning_word = ""
    for word in word_list:
        word_score = score_word(word)
        if word_score > highest_score:
            highest_score = word_score
            winning_word = word
        elif word_score == highest_score:
            if len(word) == 10 and len(winning_word) != 10:
                winning_word = word
            elif len(word) < len(winning_word) and len(winning_word) != 10:
                winning_word = word
    return (winning_word, highest_score)
```
Each iteration we run logic to see if we need to update the winning word based off if it is either scored higher than the current winning word or if it ties. If it ties then we can  see where that word falls in the tie breaking logic.